src/utils/utils_posix_common.c

```diff
@@ -104,3 +106,8 @@ umf_result_t utils_duplicate_fd(int pid, int fd_in, int *fd_out) {
     return UMF_RESULT_SUCCESS;
 #endif
 }
+
+int util_is_symlink(const char *path) {
+    struct stat st;
+    return lstat(path, &st) == 0 && S_ISLNK(st.st_mode);
```

Comment:
please create consistent return values in this function.
On windows it is -1 in case of error, and 1 in case of symlink and 0 otherwise 
on linux error in not distinguished from not a symlink